src/lib.rs

```diff
@@ -1,16 +1,26 @@
 extern crate core;
 
 use crate::error::ErrorReported;
+use std::fs::write;
 
 mod ast;
 mod codegen;
 pub mod error;
 pub mod parser;
 pub mod ty_check;
 
-pub fn compile(file_path: &str) -> Result<String, ErrorReported> {
+pub fn compile(file_path: &str, output_path: Option<&str>) -> Result<String, ErrorReported> {
```

Comment:
Before `compile` was the entry into the entire compiler. Now there is additional functionality on top. It now seems odd to me to mix the combination of the compilation stages with file access.

I suggest breaking this up:
```suggestion
pub fn compile(source: &SourceCode) -> Result<String, ErrorReported> {
  // return the generated code or errors
}
```
I am ignoring that the error reporting should probably be changed according to PR #31.

This way the implementation is more modular. For now, the CLI could open the file and decide whether to write the output to a file or to the standard output stream.

Reply:
I refactored my implementation on that point. In the changes I just pushed, the `compile` function remains unchanged, and all file-handling logic has been moved to `main.rs`.